### DISCUSSION TITLE
Integration: LlamaIndex JS/TS

### DISCUSSION THREAD

**Discussion by marcklingen** [langfuse/langfuse#1291] [Ideas]
### Describe the feature or potential improvement

Since Llamaindex Python was super popular (discussion [here](https://github.com/orgs/langfuse/discussions/828)), let's see how much interest there is in an integration with LlamaIndex JS/TS.

Feel free to add your thoughts/upvotes here.

**Comment by RajeswariKumaran** [comment]
Is it possible to use llama-index evaluators because of this integration?

**Comment by parkuman** [comment]
Integration with the TypeScript library would be huge for Langfuse in my opinion! There's a fairly popular [create-llama](https://github.com/run-llama/create-llama/) package which can get you up and running with llamaindex, Vercel AI SDK, and OpenLLMetry ([see their example template here](https://github.com/run-llama/create-llama/tree/main/templates/types/streaming/nextjs)). For example, being able to quickly get up and running creating a RAG chatbot with Langfuse tracing and deploy it straight to Vercel using NextJS / SvelteKit could bring in a big crowd of devs. 

Would be very keen in helping make this a reality.

**Reply by marcklingen** [reply]
thanks for your +1 on this!

**Comment by clemra** [comment]
+1 on twitter: https://x.com/kaitakami_/status/1836419763964199144

**Reply by kaitakami** [reply]
:) @LucianoTrujillo @sarabiaops @festevezz

**Comment by afonsomatos** [comment]
Please I realy need this.

**Comment by marcklingen** [comment]
Great to see so much interest in this integration!

By the way, we are very open for open source contributions on this. There is a python implementation including a test suite which could simplify the port to Javascript. Have not investigated this yet though as the instrumentation in LlamaIndex JS might be very different from python.
